discovery-infra/test_infra/controllers/node_controllers/vsphere_controller.py

```diff
@@ -0,0 +1,136 @@
+from builtins import list
+from typing import Tuple, List, Callable, Any
+
+import libvirt
+
+from test_infra.helper_classes.config import BaseClusterConfig
+from test_infra.helper_classes.config.vsphere_config import VSphereControllerConfig
+from test_infra.controllers.node_controllers.disk import Disk
+from test_infra.controllers.node_controllers.node import Node
+from test_infra.controllers.node_controllers.node_controller import NodeController
+from test_infra.tools import terraform_utils
+from test_infra.utils.terraform_util import TerraformControllerUtil
+
+
+class VSphereController(NodeController):
+
+    def __init__(self, config: VSphereControllerConfig, cluster_config: BaseClusterConfig):
+        super().__init__(config, cluster_config)
+        self.cluster_name = cluster_config.cluster_name.get()
+        folder = TerraformControllerUtil.create_folder(self.cluster_name, platform=cluster_config.platform)
+        self._tf = terraform_utils.TerraformUtils(working_dir=folder)
+
+    def start_all_nodes(self) -> List[Node]:
+        self._tf.set_and_apply(**self._config.get_all(), **self._cluster_config.get_all(),
+                               cluster_name=self.cluster_name)
+        return self.list_nodes()
+
+    def list_nodes(self) -> List[Node]:
+        vms_object_type = self._tf.get_resources(resource_type="vsphere_virtual_machine")
+
+        if vms_object_type:
+            return list()
+
+        vms = vms_object_type[0].instances
+
+        def vsphere_vm_to_node(terraform_vm_state):
+            return Node(name=terraform_vm_state["name"], private_ssh_key_path=self._config.private_ssh_key_path,
+                        node_controller=self)
+
+        return list(map(vsphere_vm_to_node, vms))
+
+    def destroy_all_nodes(self) -> None:
+        self._tf.destroy()
+
+    def shutdown_node(self, node_name: str) -> None:
+        pass
```

Comment:
```suggestion
        raise NotImplementedError
```
and the same for all other non implemented methods